Frameworks/Foundation/NSOperation.mm

```diff
@@ -77,41 +64,85 @@ + (BOOL)automaticallyNotifiesObserversForKey:(NSString*)key {
     return NO;
 }
 
-/**
- @Status Interoperable
-*/
-+ (id)allocWithZone:(NSZone*)zone {
-    NSOperation* ret = [super allocWithZone:zone];
+- (void)_checkReady {
```

Comment:
>_checkReady [](start = 8, length = 11)

_checkReady can be changed to O(1) instead of O(N) by keeping the ready count.  since you are observing dependencies and their 'finished' state, incrementing/decrementing the counter should do the trick.